src/svm/programConnectors.ts

```diff
@@ -0,0 +1,31 @@
+import { Idl, Program, Provider } from "@coral-xyz/anchor";
+import {
+  SvmSpokeIdl,
+  SvmSpokeAnchor,
+  MessageTransmitterAnchor,
+  MessageTransmitterIdl,
+  TokenMessengerMinterAnchor,
+  TokenMessengerMinterIdl,
+  MulticallHandlerAnchor,
+  MulticallHandlerIdl,
+} from "./assets";
+
+export function getConnectedProgram<P extends Idl>(idl: P, provider: Provider) {
+  return new Program<P>(idl, provider);
```

Comment:
Note: In the SVM scripts, we have been connecting to the program ID defined in the IDL file. Since these IDs are generated automatically, any future changes to the program, if merged into master, could cause the program ID to drift.
We should explore alternatives where the caller can pass a specific program ID in a followup PR so we can use this in our bots to prevent this unexpected programIDs changes. Wdyt?

Reply:
agree, we should better implement swapping out the programId

Reply:
Finally implemented the programId override as it enhances the usability of these functions

Reply:
ye, there are similar things we'd want to do on the evm side I think wherein the deployments/release process also includes a hash or some deterministic way to derive the code we are revering to on a release.

can fit into this as well.